File: src/test/scala/MatrixGenRecordReaderTest.scala
import org.apache.hadoop.io._
import org.junit.Test
import org.scalatest._
import org.scalatest.mockito.MockitoSugar
import org.unipi.matrixgen.{FakeInputSplit, MatrixGenRecordReader, MatrixGenerator}
import org.unipi.matrixnorm.HadoopMatrixNorm_

class MatrixGenRecordReaderTest extends FlatSpec with MockitoSugar {

  @Test
  def wrongInput(): Unit = {
    val reader = new MatrixGenRecordReader
    val mapper = new HadoopMatrixNorm_.MatrixNormMapper
    val context = mock[mapper.Context]
    val mg = new MatrixGenerator

    var progress = 0.0
    var i = 0
    while( reader.nextKeyValue()) {
      assert(reader.getCurrentKey.getClass == (new Text).getClass, true)
      assert(reader.getCurrentValue.getClass == NullWritable.get.getClass, true)
      assert(reader.getProgress > progress, true)
      progress = reader.getProgress
      val matrix = mg.deserialize(reader.getCurrentKey.toString)
      assert(matrix.nonEmpty, true)
      assert(matrix.head.nonEmpty, true)
      i += 1
    }

    assert(100 == i, true)
  }

}

File: src/test/java/MatrixGenRecordReaderTest.java
import org.unipi.matrixnorm.Utils;
import org.unipi.matrixgen.MatrixGenRecordReader;

import org.apache.hadoop.io.Text;
import org.apache.hadoop.io.NullWritable;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class MatrixGenRecordReaderTest {

    @Test
    public void wrongInput() throws IOException, InterruptedException {
        MatrixGenRecordReader reader = new MatrixGenRecordReader();

        double progress = 0.0;
        int i = 0;
        while (reader.nextKeyValue()) {
            assertEquals(reader.getCurrentKey().getClass(), Text.class);
            assertEquals(reader.getCurrentValue().getClass(), NullWritable.get().getClass());
            assertTrue(reader.getProgress() > progress);
            progress = reader.getProgress();
            Double[][] matrix = Utils.deserialize(reader.getCurrentKey().toString());
            assertTrue(matrix.length > 0);
            assertTrue(matrix[0].length > 0);
            i += 1;
        }

        assertTrue(100 == i);
    }
}
